src/storage/store/engine.rs

```diff
@@ -1080,6 +1210,12 @@ impl ShardEngine {
                 if user_data.r#type == proto::UserDataType::Username as i32 {
                     self.validate_username(message_data.fid, &user_data.value, txn_batch)?;
                 }
+                if user_data.r#type == proto::UserDataType::UserDataPrimaryAddressEthereum as i32 {
+                    self.validate_ethereum_address_ownership(message_data.fid, &user_data.value)?;
+                }
+                if user_data.r#type == proto::UserDataType::UserDataPrimaryAddressSolana as i32 {
+                    self.validate_solana_address_ownership(message_data.fid, &user_data.value)?;
```

Comment:
I would have these call `verify_fid_owns_address` directly with the the appropriate protocol. Don't convert the address, just convert it to bytes and just check that it matches the verification exactly.

Reply:
I don't think this would work if I called `verify_fid_owns_address`  directly because doing so I would be comparing the bytes of an encoded address (`"0x00000...".bytes()`) with the bytes of a decoded address, right?

Reply:
- user_data_body.value.as_bytes() - this gives us the UTF-8 bytes of the checksummed string 
  - deleted_verification.address - this is the raw address bytes

Reply:
Sample test: https://github.com/farcasterxyz/snapchain/pull/465/files#diff-6524ecb933c89497277e430bf140343f1a2d597c2001b24993ff1dd60cbcffc2R697